examples/csharp/HelloPhi/Program.cs

```diff
@@ -143,4 +144,23 @@ void PrintUsage()
         var totalTokens = outputSequence.Length;
         Console.WriteLine($"Streaming Tokens: {totalTokens} Time: {runTimeInSeconds:0.00} Tokens per second: {totalTokens / runTimeInSeconds:0.00}");
     }
+    else if (option == 3) //Streaming Chat
+    {
+        using var tokenizerStream = tokenizer.CreateStream();
+        using var generator = new Generator(model, generatorParams);
+        do
+        {
+            generator.AppendTokenSequences(sequences);
+            while (!generator.IsDone())
+            {
+                generator.GenerateNextToken();
+                Console.Write(tokenizerStream.Decode(generator.GetSequence(0)[^1]));
+            }
+            Console.WriteLine();
+            Console.WriteLine("Next prompt:");
+            prompt = Console.ReadLine();
+            sequences = tokenizer.Encode($"<|user|>{prompt}<|end|><|assistant|>");
+        } while (prompt != null);
```

Comment:
Could we avoid writing a while loop within another while loop? To get out of the while loop, we need to exit both the while loops. And it seems like the purpose for both the while loops is the same.

Reply:
Actually the outer while loop is a new conversation